File: backend_go/internal/dto/course.go
package dto

type CourseListItem struct {
	ID          string `json:"id"`
	Title       string `json:"title"`
	Description string `json:"description"`
	Image       string `json:"image"`
	Status      string `json:"status,omitempty"`
}

type CourseListResponse struct {
	Courses []CourseListItem `json:"courses"`
	Total   int              `json:"total"`
}

type CourseDetailResponse struct {
	Course CourseDetail `json:"course"`
}

type CourseDetail struct {
	ID          string             `json:"id"`
	Title       string             `json:"title"`
	Breadcrumb  string             `json:"breadcrumb"`
	Description string             `json:"description"`
	Image       string             `json:"image"`
	Status      string             `json:"status,omitempty"`
	Sections    []MaterialSection  `json:"sections"`
	Questions   []QuizQuestionItem `json:"questions"`
}

type MaterialSection struct {
	Type    string `json:"type"`
	Title   string `json:"title"`
	Content string `json:"content,omitempty"`
	URL     string `json:"url,omitempty"`
}

File: gestory/src/types/course.ts
/**
 * Course Types
 * Contracts untuk course-related data dari backend
 */

export interface MaterialSection {
  type: "pdf" | "text" | "video" | "quiz";
  title: string;
  content?: string;
  url?: string;
}

export interface CourseDetail {
  id: string;
  title: string;
  breadcrumb: string;
  description: string;
  image: string;
  status?: string;
  sections: MaterialSection[];
  questions: QuizQuestion[];
}

export interface CourseListItem {
  id: string;
  title: string;
  description: string;
  image: string;
  status?: string;
}

export interface QuizQuestion {
  id: string;
  question: string;
  options: Record<string, string>;
  correctAnswer: string;
  explanation: string;
}

export interface CourseListResponse {
  courses: CourseListItem[];
  total: number;
}

export interface CourseDetailResponse {
  course: CourseDetail;
}

/**
 * Extend dari existing interfaces di src/data/courses.ts
 * Sekarang sudah di-centralize di sini
 */
